firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/Personalization.java

```diff
@@ -58,9 +74,29 @@ public void logArmActive(@NonNull String key, @NonNull ConfigContainer configCon
       return;
     }
 
+    String personalizationId = metadata.optString(PERSONALIZATION_ID);
+    if (personalizationId.isEmpty()) {
+      return;
+    }
+
+    synchronized (armsCache) {
+      if (armsCache.get(key) == personalizationId) {
+        return;
+      }
+      armsCache.put(key, personalizationId);
+    }
+
     Bundle params = new Bundle();
-    params.putString(ARM_KEY, metadata.optString(PERSONALIZATION_ID));
+    params.putString(ARM_KEY, key);
     params.putString(ARM_VALUE, values.optString(key));
+    params.putString(PERSONALIZATION_ID_KEY, personalizationId);
+    params.putInt(ARM_INDEX_KEY, metadata.optInt(ARM_INDEX, -1));
+    params.putString(GROUP, metadata.optString(GROUP));
     analyticsConnector.logEvent(ANALYTICS_ORIGIN_PERSONALIZATION, ANALYTICS_PULL_EVENT, params);
```

Comment:
Can these log statements fail? Are we okay with one succeeding if the other doesn't?

Reply:
I think it should be OK. As long as it's not happening too frequently, the stats should be close enough.